packages/vhd-lib/Vhd/VhdAbstract.js

```diff
@@ -334,4 +334,25 @@ exports.VhdAbstract = class VhdAbstract {
     stream.length = footer.currentSize
     return stream
   }
+
+  /*
+   * check if all the data of a child are already contained in this vhd
+   */
+
+  async contains(child) {
+    await this.readBlockAllocationTable()
+    await child.readBlockAllocationTable()
+    for await (const block of child.blocks()) {
```

Comment:
Don't read blocks before checking the parent contains them.

Maybe we should add another iterator (`blockIds()`)?